inst/include/Rcpp/api/meat/Rcpp_eval.h

```diff
@@ -21,36 +21,36 @@
 #include <Rcpp/Interrupt.h>
 #include <Rversion.h>
 
-// outer definition from RcppCommon.h
-#if defined(RCPP_USE_UNWIND_PROTECT)
-  #if (defined(RCPP_PROTECTED_EVAL) && defined(R_VERSION) && R_VERSION >= R_Version(3, 5, 0))
-     // file-local and only used here 
-     #define RCPP_USE_PROTECT_UNWIND
-  #endif
+#if (defined(RCPP_PROTECTED_EVAL) && defined(R_VERSION) && R_VERSION >= R_Version(3, 5, 0))
+#define RCPP_USE_PROTECT_UNWIND
+#include <csetjmp>
 #endif
 
+
 namespace Rcpp {
 namespace internal {
 
 #ifdef RCPP_USE_PROTECT_UNWIND
 
+    // Store the jump buffer as a static variable in function scope
+    // because inline variables are a C++17 extension.
+    inline std::jmp_buf* get_jmpbuf_ptr() {
+        static std::jmp_buf jmpbuf;
+        return &jmpbuf;
+    }
+
     struct EvalData {
         SEXP expr;
         SEXP env;
         EvalData(SEXP expr_, SEXP env_) : expr(expr_), env(env_) { }
     };
 
+    // First jump back to the protected context with a C longjmp because
```

Comment:
Nice! Did you already test this on 32bit Windows? I bet this fixes the crash there (cc: @jeroen, since we were looking at this before and it looks like @lionel- has a nice solution)

Reply:
Are the i386 checks performed by R CMD check sufficient? Or should I set up another virtual box with a 32 bit Windows to check that architecture properly?

Reply:
Good point, I think that will be sufficient. I also validated this works as expected on my own VM. Nice!

One other thought: do we need to worry about this interfering with other ways that R might jump (e.g. interrupts, browser)? I think we discussed this already and we're pretty sure that we'll do the right thing as long as `R_ContinueUnwind()` is called but curious if you've already thought through these scenarios.

Reply:
> do we need to worry about this interfering with other ways that R might jump (e.g. interrupts, browser)?

All sorts of longjumps are covered by protect-unwind.

I've been wondering about longjumps occurring in destructors during a LongjumpException unwind. According to R semantics the last longjump should win but in C++ it's not possible to rethrow from a destructor. I think if C++11's `current_exception()` always returned a reference to the current exception we could update the token but unfortunately this is implementation-dependent and the `exception_ptr` might point to a copy.

So I think code in destructors that might cause longjumps should be unwind-protected and guarded with `catch(...)`, and the last longjump should simply be discarded. Too bad that exception handling in C++ is so wonky.

Reply:
> unfortunately this is implementation-dependent and the exception_ptr might point to a copy.

oh hmm, even if the exception is copied it will still point to the same token SEXP, so we could just memcopy all the RAWSXP data from one token into the other. But then the structure of the token should become part of the API.

It may not be worth worrying about this too much as it's unlikely that complex R code will be run from destructors?

Reply:
That makes good sense, thanks! We used to have complex R code in a dplyr dtor but that is no longer the case IIRC.